### PR TITLE
Django guide + python updates.

### PR DESCRIPTION
**Pull request by chadwcarlson**
## Why

Partially addresses https://github.com/platformsh/platformsh-docs/issues/2832

### General

- Extends the Django stack guide
- Adds pages to the shared Python documentation
- Reorganizes the Development/Sanitize databases section to include a Django + PostgreSQL example with the same format, adding supporting shortcodes for shared content.

### Details 

- `docs/data`: `django4` has been added to data files to support pulling template configuration into `guides/django/deploy`
- `docs/layouts/shortcodes/`
- `docs/src/development/local/ddev`: content sections that could be reused within individual stack guides -Django and future - moved to shortcodes.
      - some supporting shortcodes used through Django added for servers, local development, and database sanitization.
      - some content within the previous DDEV documentation moved to shortcodes, so it can be reused in Stack guides and the main DDEV doc page.
- `docs/src/development/local/sanitize-db`: Singe MariaDB + Drupal/drush example made into a directory. Previous text placed on list page, MariaDB as example 1, Django + PostgreSQL newly added example 2. 
- `docs/src/guides/django/deploy`: "Get started" guide for Django expanded to match current Stack format.
- `docs/srcs/guides/django/local`: Local development stack guide created for Django. Includes DDEV, Docker Compose, and Tethered local examples - also replicating the `sectionBefore` option used in the main `development/local` docs.
- `docs/src/languages/python` - expanded to include 1) a page that addresses using pip, pipenv and poetry, 2) a page that addresses using production web servers outside of what's shown in the dedicated stack guide. 
- `themes/psh-docs/layouts/partials/note.html`: small spacing issue on notes fixed.

## TODO

- [x] Address sanitize-db merge conflict
- [x] Address failing vale tests in the pipeline
- [x] Remove vale exceptions that shouldn't be caught within codetabs on Aaron's advice
- [x] Track back no longer needed data/example file pulls to match `main`
- [x] `/guides/django` local development instructions
- [x] Update `/guides/django/deploy/next-steps.html` to link back to shared python resource pages.

All other tasks in https://github.com/platformsh/platformsh-docs/issues/2832 can be addressed in separate PRs.

### CODE REVIEW

**Comment by chadwcarlson** [comment]
The different versions of `init-local.sh` are not needed here https://python-cgskmcy-652soceglkw4u.eu-3.platformsh.site/guides/django/local/docker-compose.html#onboard-collaborators

File: docs/src/create-apps/upgrading.md [hunk 1]
```diff
@@ -278,7 +278,7 @@ the configuration files format. Most of the old configuration format is
 still supported, but customers are invited to move to the new format.
 
 For an example upgrade path, see the [Drupal 7.x branch of the
-Platformsh-examples
+`platformsh-examples`
```
Comment (chadwcarlson) [inline docs/src/create-apps/upgrading.md:281]: This spelling triggered a Vale failure. Let me know if you'd like to include as is, place an exception somewhere else, or address in some other way.

**Comment by chadwcarlson** [comment]
> Remove vale exceptions that shouldn't be caught within codetabs on Aaron's advice

Remove the following from [`accept.txt`](https://github.com/platformsh/platformsh-docs/blob/3e06662c43be7cd3d46c1255bc2bb564dedd4fc5/styles/Vocab/Platform/accept.txt):

- collectstatic
- ddev
- depends_on
- env
- ENV
- gunicorn
- healthcheck
- hypercorn
- mv
- passthru
- php_version
- pipenv
- platformsh
- postgres
- printf
- psql
- runserver
- socket_family
- source_path
- tmpfile
- unix
- uvicorn

**Comment by chadwcarlson** [comment]
@CollierCZ Thanks for the review! I'm fine moving these other instructions to the community site. 

To be clear, would you like me to remove just Docker Compose, or also Tethered?

**Comment by CollierCZ** [comment]
That's a good point. I think Tethered can make sense to keep, even if not recommended. So just the Docker Compose page. Thanks!

**Comment by chadwcarlson** [comment]
@CollierCZ Moved and linked here https://community.platform.sh/t/using-docker-compose-with-django/1205 👍

**Review by CollierCZ — APPROVED** [review]
:rocket:

**Pull request merged**